### PR TITLE
updated examples and added purchasePlan

### PR DESCRIPTION
**Pull request by meet1993shah**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json [hunk 1]
```diff
@@ -2800,6 +2834,10 @@
             "modelAsString": true
           }
         },
+        "purchasePlan": {
+          "$ref": "#/definitions/PurchasePlan",
+          "description": "Purchase Plan Information of the Virtual Machine persisted on the Disk."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json]: Purchase Plan Information of the image used to create the Disk?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json [hunk 2]
```diff
@@ -1800,6 +1804,10 @@
...
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json]: if this is snapshot say that purchase plan information copied from source disk/snapshot

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 1 above)
Comment (meet1993shah) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json]: > Purchase Plan Information of the image used to create the Disk?

Optional field, if the Customer wants to put specific Purchase Plan then they can.

Comment (chetmaddula) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json]: might need a better description, please ask Vikram about it. Thanks

Comment (vikramd-ms) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json]: Purchase plan information for the the image from which the OS disk was created.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 2 above)
Comment (vikramd-ms) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json]: Purchase plan information for the image from which the source disk for the snapshot was originally created.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json (hunk 1 above)
Comment (vikramd-ms) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json]: add examples

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json [hunk 3]
```diff
@@ -25,6 +25,12 @@
         "properties": {
           "provisioningState": "Updating",
           "osType": "Windows",
+          "purchasePlan": {
+            "name": "{VMPurchasePlanName}",
+            "publisher": "{VMPurchasePlanPublisherId}",
+            "product": "{VMPurchasePlanProduct}",
+            "PoromotionCode": "{VMPurchasePlanPromotionCode}"
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json]: typo. Examples should be generated by running tests

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json [hunk 4]
```diff
@@ -41,6 +47,12 @@
...
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json]: typo

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json [hunk 5]
```diff
@@ -17,6 +17,12 @@
           "creationData": {
             "createOption": "Empty"
           },
+          "purchasePlan": {
+            "name": "{VMPurchasePlanName}",
+            "publisher": "{VMPurchasePlanPublisherId}",
+            "product": "{VMPurchasePlanProduct}",
+            "PoromotionCode": "{VMPurchasePlanPromotionCode}"
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json]: typo

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json [hunk 6]
```diff
@@ -39,12 +46,18 @@
         "name": "myDisk",
         "location": "West US",
         "properties": {
-          "provisioningState": "Updating",
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json:42]: is promotionCode missing?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json [hunk 7]
```diff
@@ -1,56 +1,48 @@
 {
   "parameters": {
-    "subscriptionId": "{subscription-id}",
-    "resourceGroupName": "myResourceGroup",
+    "subscriptionId": "850f02f7-e14c-4fb7-905d-7b98bcae9c7f",
+    "resourceGroupName": "meeshahtest",
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json]: give a generic rg and subscription id..

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutASnapshot.json [hunk 8]
```diff
@@ -13,6 +13,12 @@
           "creationData": {
             "createOption": "Empty"
           },
+          "purchasePlan": {
+            "name": "{VMPurchasePlanName}",
+            "publisher": "{VMPurchasePlanPublisherId}",
+            "product": "{VMPurchasePlanProduct}",
+            "PoromotionCode": "{VMPurchasePlanPromotionCode}"
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutASnapshot.json]: typo: promotionCode

**Review by chetmaddula — CHANGES_REQUESTED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json (hunk 6 above)
Comment (meet1993shah) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json:42]: its optional, some purchase plan have it and some dont, I am not able to find the one where its present.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json [hunk 9]
```diff
@@ -3,54 +3,46 @@
...
+          "diskMBpsReadWrite": 60,
           "encryption": {
             "type": "EncryptionAtRestWithPlatformKey"
           },
-          "timeCreated": "2016-12-28T04:41:35.079872+00:00",
-          "provisioningState": "Succeeded"
-        },
-        "type": "Microsoft.Compute/disks",
-        "location": "westus",
-        "tags": {
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json]: tags are missing

Comment (meet1993shah) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json]: created the disk without tags, do I need to add tags?

Comment (chetmaddula) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json]: yes, its supposed to show all the possible info in get disk only new properties need to be added, existing ones shouldnt be removed

Comment (meet1993shah) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json]: oh I see okay, will update it.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json [hunk 10]
```diff
@@ -11,7 +11,7 @@
         "creationData": {
           "createOption": "FromImage",
           "imageReference": {
-            "id": "/Subscriptions/{subscriptionId}/Providers/Microsoft.Compute/Locations/uswest/Publishers/Microsoft/ArtifactTypes/VMImage/Offers/{offer}"
+            "id": "/Subscriptions/{subscriptionId}/Providers/Microsoft.Compute/Locations/uswest/Publishers/{publisher}/ArtifactTypes/VMImage/Offers/{offer}/Skus/{sku}/Versions/1.0.0"
```
Comment (laurawu19) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json]: Locations/uswest? or Locations/westus? Respond should be the same as request.

**Review by laurawu19 — CHANGES_REQUESTED** [review]

**Review by laurawu19 — APPROVED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/UpdateAManagedDiskToAddPurchasePlan.json [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+          "creationData": {
+            "createOption": "FromImage",
+            "imageReference": {
+              "id": "/Subscriptions/{subscription-id}/Providers/Microsoft.Compute/Locations/westus/Publishers/test_test_pmc2pc1/ArtifactTypes/VMImage/Offers/marketplace_vm_test/Skus/test_sku/Versions/1.0.0"
+            }
+          },
+          "diskSizeGB": 127,
+          "provisioningState": "Succeeded",
+          "isArmResource": true,
+          "faultDomain": 0
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/UpdateAManagedDiskToAddPurchasePlan.json]: isArmResource and faultDomain are internal only properties and should not appear in the response

**Review by vikramd-ms — APPROVED** [review]

**Pull request merged**